### ISSUE TITLE
Deal with popup that exceeds maxBounds. (Edit)

### ISSUE THREAD

**Issue by danallsop** [ScottLogic/jumpstart-Leaflet#72]
> **Migrated from:** https://github.com/Leaflet/Leaflet/issues/2324
> **Original author:** danallsop
> **Original created:** 2013-12-20T18:07:52Z
---

If you open a popup that exceeds the bounds defined via the maxBounds option then the popup will be cut on top. There should be some mechanism that offers a easy way to deal with this problem. I can imagine:
- when testing for bounds limit include the popup size
- render the popup differently (on the bottom, left or right) if it exceeds the maxBounds.
- Add mode that lets popups break out of the map viewport.

**Comment by danallsop** [comment]
## Comments (14 total)

---
> **Comment by:** danallsop **on** 2013-12-25T18:23:36Z

Yeah, will check this out.

---
> **Comment by:** danallsop **on** 2014-01-17T05:34:36Z

I've experienced the same issue.  The popup is displayed and the map attempts to pan to view all of the popup's contents, but then bounces back into the area restricted by maxBounds. 

---
> **Comment by:** danallsop **on** 2014-06-29T09:25:45Z

Or, we can stuff the content into a <p> element, and make `overflow: scroll`, So we can scroll inside the popup box.

---
> **Comment by:** danallsop **on** 2014-11-05T20:14:45Z

Quick note to say this affects me as well. I thought I could get around it by adding padding to my maxBounds but somehow this is giving me a lot of trouble. Hoping the next leaflet version fixes this.

---
> **Comment by:** danallsop **on** 2015-05-04T10:04:59Z

+1 I think this is crucial when working with image overlays

---
> **Comment by:** danallsop **on** 2015-05-08T15:01:20Z

I think this is covered by the Rrose plugin (http://erictheise.github.io/rrose/), so there would be no need to implement this in the core.

---
> **Comment by:** danallsop **on** 2015-05-08T15:19:23Z

Well, it's not super-critical but we'd still want to fix that because it's not just a feature, but a bug you can trigger quite easily and that harms accessibility.

---
> **Comment by:** danallsop **on** 2015-06-03T11:05:16Z

Anyone able to look at this pretty please? :+1: 

---
> **Comment by:** danallsop **on** 2015-06-03T11:21:56Z

danallsop just use the RRose plugin for now.

---
> **Comment by:** danallsop **on** 2015-06-03T11:26:29Z

danallsop I've looked at it but was unsure if I can get it working with https://github.com/tombatossals/angular-leaflet-directive or not - I will give it a try.

---
> **Comment by:** danallsop **on** 2017-10-17T16:09:09Z

Not sure if my issue is similar enough but:
FlyTo() does not seem to respect maxBounds. 

When I animate focusin on one marker on my map (which is inside bounds), then flyTo will still move the map to have said marker centered. Then the map bounces back to fix itself according to max bounds. It would be great if the function could calculate an offset for the animation when maxbounds are set
---
> **Comment by:** danallsop **on** 2017-12-11T14:03:51Z

So, two years later we still have to use Rrose - which is not as good a solution as having it fixed in the leaflet core module.
Leaflet is used as an ecosystem of third party components. It is annoying to monkey-patch Vue2Leaflet, or Angular components, with Rrose, just to give two examples.
---
> **Comment by:** danallsop **on** 2018-02-13T03:08:50Z

My suggestion is for options as follows:

1. Popup underneath if marker latitude is south of centre latitude
2. When autopanning, allow bounds to be exceeded if required to accommodate popup.   When popup closes go back to map centre as before opened .
---
> **Comment by:** danallsop **on** 2022-01-05T07:40:07Z

To stay consistent, I suggest to add a `direction` option like [there is for `Tooltip`](https://leafletjs.com/reference.html#tooltip-direction). The `auto` value then positions the popup so that it always stays within the `maxBounds`.

Or, if possible, move it to [`DivOverlay`](https://leafletjs.com/reference.html#divoverlay), so all overlays based on it can benefit from that functionality.

Sebastian
